src/tests/core.py

```diff
@@ -274,6 +276,52 @@ def test_old_knownnodes_pickle(self):
         finally:
             cleanup(files=('knownnodes.dat',))
 
+    @staticmethod
+    def generate_random_address():
+        """Generating random address"""
+        import queues
+        streamNumberForAddress = 1
+        queues.addressGeneratorQueue.put((
+            'createRandomAddress', 4, streamNumberForAddress,
+            "test1", 1, "", False))
+
+    @staticmethod
+    def delete_address_from_database(address):
+        """Clean up addressbook"""
+        sqlQuery('''delete from addressbook where address=?''', address)
+
+    def test_add_same_address_twice_in_addressbook(self):
+        """checking same address is added twice in addressbook"""
+        self.assertTrue(helper_addressbook.insert(label='test1', address=self.addr))
+        self.assertFalse(helper_addressbook.insert(label='test1', address=self.addr))
+        self.delete_address_from_database(self.addr)
+
+    def test_is_address_present_in_addressbook(self):
+        """checking is address added in addressbook or not"""
+        helper_addressbook.insert(label='test1', address=self.addr)
+        queryreturn = sqlQuery('''select count(*) from addressbook where address=?''', self.addr)
+        self.assertTrue(bool(queryreturn[0][0]))
+        self.delete_address_from_database(self.addr)
+
+    def test_is_own_address_add_to_addressbook(self):
+        """Checking own address adding in addressbook"""
+        self.generate_random_address()
+        time.sleep(.5)
```

Comment:
instead of this sleep look at how it's done in `support.py` and put the waiting into `generate_random_address` directly